dotcom-rendering/src/lib/useUserPreferredEdition.ts

```diff
@@ -0,0 +1,71 @@
+import { getCookie, removeCookie, setCookie } from '@guardian/libs';
+import { useEffect, useState } from 'react';
+import { mutate } from 'swr';
+import {
+	type EditionId as Edition,
+	getEditionFromPageId,
+	isNetworkFront,
+} from './edition';
+
+const getBannerValueFromQueryParams = () => {
+	const queryParams = new URLSearchParams(window.location.search);
+	return queryParams.get('editionSwitcherBanner');
```

Comment:
Could making this param lowercase mean less chance for mistakes? It seems to be what we have favoured in the past in Frontend:

https://github.com/guardian/frontend/blob/a6884771da35cfd63a9032400bbd59b89aef0a03/common/app/dev/DevParametersHttpRequestHandler.scala#L67

Reply:
Sigh. I thought the commercial param names were fairly consistent but it's a shame there is a mix of conventions.

Happy to leave it to you though unless you fancy `edition-switcher-banner`

As an aside the received wisdom appears to be to use hyphens for word separators in URLs:
https://developers.google.com/search/docs/crawling-indexing/url-structure
https://blog.codinghorror.com/of-spaces-underscores-and-dashes/

Reply:
OK interesting. If we're inconsistent then I think going with what Google recommend is probably a good idea. I'll update to `edition-switcher-banner`.